web-ui/src/helpers/query-parameters.js

```diff
@@ -21,8 +21,21 @@ import { useEffect } from 'react';
 /**
  * @param {(QPBoolean | QPString)[]} qps - query parameters
  */
-export const useQueryParameters = qps => {
+export const useQueryParameters = (
+  qps,
+  requirements = [],
+  processedQPs = null
+) => {
+  // This examines all the query parameters and
+  // sets the state value associated with each of them.
   useEffect(() => {
+    if (processedQPs?.current) return;
+
+    const haveRequirements = requirements.every(req =>
+      Array.isArray(req) ? req.length > 0 : req !== null && req !== undefined
```

Comment:
When a required value is an array, the requirement is met when the array is not empty.
Otherwise the requirement is met when the value is not `null` or `undefined`.